packages/server/lib/controllers/oauth.controller.ts

```diff
@@ -1505,9 +1650,11 @@ class OAuthController {
                         },
                         session.provider,
                         activityLogId,
-                        { initiateSync, runPostConnectionScript }
+                        { initiateSync, runPostConnectionScript },
+                        logCtx
                     );
                 }
+                await logCtx.success();
```

Comment:
yep agreed, there is no easy way to do unless you wrap everything with a big function which is annoying and would increase the change related by a huge amount. 
However the spec has evolved and since Bastien wants something closer to an auditlog I will probably end up provisioning the logContext at the API level and thus wrap the endpoints transparently. BUT it's not going to be that easy because some endpoints returning does not mean the activity is done (i.e: oauth login)

If you have an other idea I'm definitely open.

Reply:
yeah let's discuss